### ISSUE TITLE
Session gets invalidated frequently last time

### ISSUE THREAD

**Issue by msvprogs** [trevorhobenshield/twitter-api-client#224]
Hello,
Does someone experience the same phenomenon?
During last weeks, I get '32 - Could not authenticate you' errors frequently on various requests (liking, DM, search). I don't do anything that could be identified as abusive behavior (DM spamming, a lot of subsequent random likes, exceeding endpoint rate limits). Could this be due to missing request headers `X-Client-UUID` and `x-client-transaction-id` ? 
There's an explanation on generating the latter header: https://antibot.blog/twitter-header-part-3/ 
But the author claims that presence of this header doesn't affect the trustfulness of API requests. I don't know if it's still valid, that was in February 2024. 
Maybe there's another reason for frequent session invalidations.

**Comment by Phineas1500** [comment]
yeah, i use the twitter.search feature, and after running my code for a couple minutes i get a 200 unauthorized error. if i keep my code running with those errors, i eventually get logged out of twitter on my browser. ive heard that after enough logouts, twitter suspends your account. i think we need a way to properly send the x-client-transaction-id header.

**Comment by runetech0** [comment]
I have been facing this issue as well since months. Has any solution been found yet?
Seems like generating the x-client-transaction-id header is a bit complex because the the code is obfuscated.
Here's the info about reverse engineering this header. antibot.blog.
Hope someone can implement it.

**Comment by Phineas1500** [comment]
I know Dimden has implemented it for Old Tweetdeck. The code is in his Github, specifically in the source here: https://tweetdeck.dimden.dev/solver.html

His repo: https://github.com/dimdenGD/OldTweetDeck
